### PR TITLE
MAGN-6404 Web Request node locks up Dynamo with infinite loop.

### PR DESCRIPTION
**Pull request by ikeough**
http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6404

This PR does two things:

1. Removes the call to `OnNodeModified()` in the `BuildOutputAST` method. The placement of this call, and the fact that it triggers an execution of the graph was causing an infinite loop, and eventually a stack overflow exception.
2. Sets `EnablePeriodicUpdate = true` to support periodic running.

PTAL:
@pboyer

### CODE REVIEW

**Comment by pboyer** [comment]
LGTM - But obviously it would be great to have some kind of test to validate your work.  I'd be appreciative if you could add that.

**Comment by ikeough** [comment]
@pboyer I need one more sanity check. I've added a system test which flexes this node, including the periodic update capability. In order to track the number of runs, I've added the `EvaluationCount` property to `HomeWorkspaceModel`. I figured this would be handy when we start validating graph interactions only cause <n> executions as well. And because it's stored on the workspace, it's compliant with multiple home workspaces.

The only down side to this test is that it does actually hit google a few times. It'll break if the testing system is not connected to the internet. The underling web request is not easily mock-able without the ability for the UI node to know that it's in testing mode.

Let me know what you think.

File: test/DynamoCoreUITests/WebRequestTests.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            var ws = Model.CurrentWorkspace as HomeWorkspaceModel;
+
+            AssertEvaluationCount(ws, 1);
+
+            // Ensure that the response is valid
+            var webRequestNode = ws.FirstNodeFromWorkspace<WebRequest>();
+            Assert.NotNull(webRequestNode);
+
+            var response = GetPreviewValue(webRequestNode.GUID.ToString()).ToString();
+            Assert.True(response.Contains("Search the world's information"));
```
Comment (pboyer) [inline test/DynamoCoreUITests/WebRequestTests.cs]: You might be able to just check for an HTTP 200.

**Comment by pboyer** [comment]
Hey @ikeough, You can check for a web connection and return make the test inconclusive if you don't have one.  Let's not worry about mocking it at this point.  

http://stackoverflow.com/questions/2031824/what-is-the-best-way-to-check-for-internet-connectivity-using-net

**Pull request merged**